src/middleware/context.ts

```diff
@@ -14,6 +14,17 @@
  * limitations under the License.
  */
 
+/*
+  When users are using middleware and this client library has access to request
+  headers, we should automatically extract as much information from the headers
+  as possible. This is in addition to when the user explicitly logs an
+  HTTPRequest logEntry object.
+
+  Scenarios:
+  - Context is available from Opencensus
```

Comment:
OpenCensus or OpenTelemetry (or in other words, from W3 Context propagation protocol)

Reply:
added the comment